src/script/page/MainContent/panels/preferences/DevicesPreferences/DevicesPreferences.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React, {useEffect, useState} from 'react';
+
+import {QualifiedId} from '@wireapp/api-client/lib/user';
+import {WireIdentity} from '@wireapp/core-crypto/platforms/web/corecrypto';
+import {container} from 'tsyringe';
+
+import {ClientEntity} from 'src/script/client/ClientEntity';
+import {CryptographyRepository} from 'src/script/cryptography/CryptographyRepository';
+import {Conversation} from 'src/script/entity/Conversation';
+import {User} from 'src/script/entity/User';
+import {useKoSubscribableChildren} from 'Util/ComponentUtil';
+import {t} from 'Util/LocalizerUtil';
+
+import {DetailedDevice} from './components/DetailedDevice';
+import {Device} from './components/Device';
+import {DeviceDetailsPreferences} from './components/DeviceDetailsPreferences';
+
+import {ClientState} from '../../../../../client/ClientState';
+import {isMLSConversation} from '../../../../../conversation/ConversationSelectors';
+import {ConversationState} from '../../../../../conversation/ConversationState';
+import {Core} from '../../../../../service/CoreSingleton';
+import {PreferencesPage} from '../components/PreferencesPage';
+
+interface DevicesPreferencesProps {
+  core?: Core;
+  clientState: ClientState;
+  conversationState: ConversationState;
+  cryptographyRepository: CryptographyRepository;
+  selfUser: User;
+  removeDevice: (device: ClientEntity) => Promise<unknown>;
+  resetSession: (userId: QualifiedId, device: ClientEntity, conversation: Conversation) => Promise<void>;
+  verifyDevice: (userId: QualifiedId, device: ClientEntity, isVerified: boolean) => void;
+}
+
+export const DevicesPreferences: React.FC<DevicesPreferencesProps> = ({
+  core = container.resolve(Core),
```

Comment:
Do we really need to inject core as a whole? It seems we use e2eiService only. Could we inject this service only? `E2EIdentity.ts` class seems to have access to core already, maybe this one can be used?